pkg/storage/util.go

```diff
@@ -56,6 +58,10 @@ func NamespaceKeyFunc(prefix string, obj runtime.Object) (string, error) {
 	if err != nil {
 		return "", err
 	}
+	name := meta.Name()
+	if ok, msg := validation.ValidatePathSegmentName(name, false); !ok {
```

Comment:
Object strategies should definitely validate names themselves (like most are already doing). This should be the minimal validation needed to ensure the REST and storage layers can address the stored name